stock_release_channel/models/stock_picking.py

```diff
@@ -103,27 +104,43 @@ def _find_release_channel_possible_candidate(self):
         self.ensure_one()
         return (
             self.env["stock.release.channel"]
-            .search(self._get_release_channel_possible_candidate_domain())
+            .search(self._release_channel_possible_candidate_domain)
             .sorted(key=lambda r: (not bool(r.partner_ids), r.sequence))
         )
 
-    def _get_release_channel_possible_candidate_domain_channel(self):
-        """Domain for finding channel candidates based on channel.
-
-        The condition is defined by the channel.
-        """
-        return [
+    @property
+    def _release_channel_possible_candidate_domain(self):
+        """Domain for finding channel candidates"""
+        self.ensure_one()
+        domain_base = self._release_channel_possible_candidate_domain_base
+        domain = [
             ("is_manual_assignment", "=", False),
-            ("state", "!=", "asleep"),
+            ("state", "in", ("open", "locked")),
             "|",
             ("picking_type_ids", "=", False),
             ("picking_type_ids", "in", self.picking_type_id.ids),
         ]
+        domain_partner = (
+            self.partner_id._release_channel_possible_candidate_domain
+            if self.partner_id
+            else []
+        )
+        domain_extras = []
+        if self._release_channel_possible_candidate_domain_apply_extras:
+            domain_extras = self._release_channel_possible_candidate_domain_extras
+        domain = expression.AND([domain, domain_base, domain_partner] + domain_extras)
```

Comment:
Could this be simplified as well?
Is the property `_release_channel_possible_candidate_domain_apply_extras` really needed. 
If you don't want to apply the extra domain, (what isn't the case right now in any depending addon) you could just overwrite `_release_channel_possible_candidate_domain_extras` and return `[]`.

```suggestion
        domain_extras = self._release_channel_possible_candidate_domain_extras
        domain = expression.AND([domain, domain_base, domain_partner] + domain_extras)
```

Reply:
There are modules extending the domains like delivery window and public holidays for checking the date is a valid customer date
And there is the sale module that bypasses those domains when the commitment date is set https://github.com/OCA/wms/pull/1036/files
As those modules do not depend on each other, I can't have one setting the domain to empty as it could be before the other modules extending the domain

Reply:
Thx. I didnt checked your other PRs